src/controller/CHIPDeviceController.cpp

```diff
@@ -3354,6 +3560,40 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+
+    case CommissioningStage::kJFValidateNOC:
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
```

Comment:
This is doing implicit fallthroughs to kSendOpCertSigningRequest when CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC is not defined.  Shouldn't it `CommissioningStageComplete` with some error instead?